### ISSUE TITLE
itv.com: ads

### ISSUE THREAD

**Issue by leahcornelius** [uBlockOrigin/uAssets#24336]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL address of the web page

`https://www.itv.com/watch/mr-robot/10a3807/10a3807a0001`

### Category

ads

### Description

Ad breaks still present with ITV X show. Not sure if this is expected but told on reddit posts your extension blocks them

### Other extensions used

Bitwarden
Tab Tree
Link to text fragment
Mino (formerly minty) - automatic coupons
Tab groups

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.58.0
Firefox: 127
filterset (summary):
 network: 137084
 cosmetic: 47000
 scriptlet: 20018
 html: 1851
listset (total-discarded, last-updated):
 default:
  user-filters: 1-0, never
  ublock-filters: 39070-114, 7m
  ublock-badware: 9080-0, 7m
  ublock-privacy: 938-2, 7m
  ublock-unbreak: 2425-2, 7m
  ublock-quick-fixes: 116-5, 7m
  easylist: 86338-179, 7m
  easyprivacy: 51116-64, 7m
  urlhaus-1: 14835-0, 7m
  plowe-0: 3543-988, 7m
filterset (user): [array of 1 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 1525 ms
 maxAssetCacheWait: 73 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 6
 network:
  itv.com: 3
  conviva.com: 2
  google.com: 1
```
</details>

**Comment by stephenhawk8054** [comment]
I can't reproduce. Test again in a new Firefox profile with only uBO and its default settings. If the issue is still there in the new profile, can you screenshot it?

**Comment by TEPJ101** [comment]
Hi, 

I'm not the original user but have ran into the same issue. I mainly use Chrome and on every profile I have tried it still blocks it with the same warning, even with only the default settings. I installed Firefox on my device and installed the extension again on it and ran into the same issue despite it being a clean install. It seems that ITVx continues to detect

**Comment by 0bsidianSn0w** [comment]
Hello! I'm afraid I have had an issue with the adverts still being played, although it is not telling me that I need to disable it. It is on a base install of Firefox and I'm pretty sure that none of the base settings gave been changed

**Comment by gothic-bum** [comment]
@0bsidianSn0w 

Please test again using a [new Firefox profile](https://support.mozilla.org/en-US/kb/profile-manager-create-and-remove-firefox-profiles) with only uBO and its default settings.

**Comment by Nimras186** [comment]
I can reproduce this as well.

I tested in a dedicated Firefox profile created specifically for uBlock Origin testing. It is not used for normal browsing and has no other extensions installed.

Before testing today I:
- removed the existing uBO installation
- restarted Firefox
- freshly installed uBlock Origin 1.74.0
- manually updated all default filter lists in the Dashboard
- confirmed there are no user filters, user setting overrides, or hidden settings

Firefox: 155
uBO: 1.74.0

On ITVX I started Lost with the uBO Logger already open.

Episode 1 played without adverts. I then moved normally to episode 2 without changing any settings, and an intro advert immediately played.

The current ITV filter was loaded:

##+js(xml-prune, VAST > Ad, '', /tserver)

During the failing intro advert the Logger showed the VAST WebSocket:

wss://vast.prd.mp.itv.com/s/v1/t

and the xml-prune scriptlet reported:

[www.itv.com][xml-prune ⁝ VAST > Ad ⁝  ⁝ /tserver] Element.Ad removed
[www.itv.com][xml-prune ⁝ VAST > Ad ⁝  ⁝ /tserver] Removing 1 items

Despite that, ITV requested and played a separate media file matching:

`https://ar.prd.content.itv.com/media/REDACTED_CTV01.HD.3500.mp4`

So in this reproduction the VAST xml-prune scriptlet did execute and remove an Ad element, but an intro advert still played.

I have the full Logger capture available, but it contains authentication/session data so I have not posted it publicly. I can provide further sanitised Logger details or repeat specific tests if useful.

**Comment by stephenhawk8054** [comment]
Please give exact link where you see the issue (not the home page).

**Comment by Nimras186** [comment]
> Please give exact link where you see the issue (not the home page).

The exact URL from my clean reproduction is:

https://www.itv.com/watch/lost/10a5894/10a5894a0002

I have seen the problem across both TV shows and films on ITVX, but it is intermittent rather than tied to one particular title.

I deliberately chose Lost for the uBO-only test because it was something I was not otherwise watching. Episode 1 played without adverts. I then loaded episode 2 normally, without changing any settings, and the intro advert played. That is the reproduction described in my previous comment.

In normal use I am currently watching Midsomer Murders. I can sometimes watch 1–4 episodes with no adverts, then have adverts start appearing for several episodes. Sometimes a reload fixes it, sometimes opening a new tab fixes it, and sometimes it takes several reloads/new tabs before blocking starts working again.

So I have not found a consistent title, episode or sequence that triggers it; the exact URL above is simply the clean test where I captured the failure in the Logger.
